code/modules/organs/organ_icon_ch.dm

```diff
@@ -0,0 +1,33 @@
+// override the organ icon getting proc
+// if digitigrade is set:
+// - Save original iconbase and markings vars to temp places
+// - set iconbase to digitigrade sprite file
+// - nullify markings list
+// - run original get_icon proc with changed vars
+// - once done, reset icobase & markings to default values from temp storage vars
+
+/obj/item/organ/external/get_icon(var/skeletal)
+	var/digitigrade = 0
+
+	// preferentially take digitigrade value from owner if available, THEN DNA.
+	// this allows limbs to be set properly when being printed in the bioprinter without an owner
+	// this also allows the preview mannequin to update properly because customisation topic calls don't call a DNA check
+	if(owner)
+		digitigrade = owner.digitigrade
+	else
+		digitigrade = dna.digitigrade
+
+	if( digitigrade && ( istype(src,/obj/item/organ/external/leg) || istype(src,/obj/item/organ/external/foot) ) )
+		var/IB = species.icobase
+		var/list/markingsTmp = markings
+
+		markings = null
+		species.icobase = species.icodigi
```

Comment:
This might be causing some issues and possibly play a part with client caches and stuff bugging out and crashing. All non-custom species use a single shared species datum between all mobs of the same species, and 2 users of the same species with different digi settings might have some nasty conflicts going on with their icon generation.

Reply:
I undid any editing of the species datum by this code and instead copy some of the icon getting code from the original to be performed in a different way specifically for digitigrade here.